homeassistant/components/hue.py

```diff
@@ -0,0 +1,245 @@
+SCENE_SCHEMA = vol.Schema({
+    vol.Required(ATTR_GROUP_NAME): cv.string,
+    vol.Required(ATTR_SCENE_NAME): cv.string,
+})
+
+CONFIG_INSTRUCTIONS = """
+Press the button on the bridge to register Philips Hue with Home Assistant.
+
+![Location of button on bridge](/static/images/config_philips_hue.jpg)
+"""
+
+
+def setup(hass, config):
+    """Set up the Hue platform."""
+    config = config.get(DOMAIN)
+    if config is None:
+        config = {}
+
+    if DOMAIN not in hass.data:
+        hass.data[DOMAIN] = {}
+
+    discovery.listen(
+        hass,
+        SERVICE_HUE,
+        lambda service, discovery_info:
+        bridge_discovered(hass, service, discovery_info))
+
+    bridges = config.get(CONF_BRIDGES, [])
+    if len(bridges) == 0:
+        _LOGGER.info("No bridges found in configuration")
+        return True
+
+    for bridge in bridges:
+        filename = bridge.get(CONF_FILENAME, PHUE_CONFIG_FILE)
+        allow_unreachable = bridge.get(CONF_ALLOW_UNREACHABLE,
+                                       DEFAULT_ALLOW_UNREACHABLE)
+        allow_in_emulated_hue = bridge.get(CONF_ALLOW_IN_EMULATED_HUE,
+                                           DEFAULT_ALLOW_IN_EMULATED_HUE)
+        allow_hue_groups = bridge.get(CONF_ALLOW_HUE_GROUPS,
+                                      DEFAULT_ALLOW_HUE_GROUPS)
+
+        host = bridge.get(CONF_HOST, None)
+
+        if host is None:
+            host = _find_host_from_config(hass, filename)
+
+        if host is None:
+            _LOGGER.error("No host found in configuration")
+            return False
+
+        setup_bridge(host, hass, filename, allow_unreachable,
+                     allow_in_emulated_hue, allow_hue_groups)
+
+    discovery.load_platform(hass, 'light', DOMAIN, {}, config)
+
+    return True
+
+
+def bridge_discovered(hass, service, discovery_info):
+    """Dispatcher for Hue discovery events."""
+    host = discovery_info.get('host')
+    serial = discovery_info.get('serial')
+
+    filename = 'phue-{}.conf'.format(serial)
+    setup_bridge(host, hass, filename)
+
+    discovery.load_platform(hass, 'light', DOMAIN, discovery_info, {})
+
+
+def setup_bridge(host, hass, filename=None, allow_unreachable=False,
+                 allow_in_emulated_hue=True, allow_hue_groups=True):
+    """Set up a given Hue bridge."""
+    # Only register a device once
+    if socket.gethostbyname(host) in hass.data[DOMAIN]:
+        return
+
+    bridge = HueBridge(host, hass, filename, allow_unreachable,
+                       allow_in_emulated_hue, allow_hue_groups)
+    hass.data[DOMAIN][socket.gethostbyname(host)] = bridge
+    bridge.setup()
+
+
+def _find_host_from_config(hass, filename=PHUE_CONFIG_FILE):
+    """Attempt to detect host based on existing configuration."""
+    path = hass.config.path(filename)
+
+    if not os.path.isfile(path):
+        return None
+
+    try:
+        with open(path) as inp:
+            return next(json.loads(''.join(inp)).keys().__iter__())
```

Comment:
jeeeez.

- Use `json.load(inp)` instead of converting file pointer to string.
- Call `iter(obj)` instead of `obj.__iter__()`